### PR TITLE
[receiver/azuremonitor] Add support for Managed Identity and DefaultAzureCredential authentication

### LINKED ISSUE #31268: [receiver/azuremonitor] Handle errors; don't ignore them
### Component(s)

receiver/azuremonitor

## Description

Code currently fails silently when getting clients from Azure SDK for metrics and metric definitions.

## Steps to Reproduce

Read the code called by start method of scraper. It currently holes errors when instantiating SDK clients the scraper uses.

Look at [getArmClient](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/d6068f1c01f6ade3517d87d83863ba4217a55775/receiver/azuremonitorreceiver/scraper.go#L137) and [azureScraper.GetMetricsValuesClient](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/d6068f1c01f6ade3517d87d83863ba4217a55775/receiver/azuremonitorreceiver/scraper.go#L157).

## Expected Result

Handle errors.

## Actual Result

Ignores errors.

### Collector version

v0.94.0

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")

### OpenTelemetry Collector configuration

```yaml
# N/A
```

### Log output

```shell
N/A
```

### Additional context

N/A

### PR DESCRIPTION
**Pull request by sriniketh923**
Description:
Enhances azuremonitor receiver authentication by using [managed identity](https://learn.microsoft.com/en-us/azure/developer/go/azure-sdk-authentication?tabs=bash#option-3-use-a-managed-identity) or [environment variables](https://learn.microsoft.com/en-us/azure/developer/go/azure-sdk-authentication?tabs=bash#option-1-define-environment-variables). I have extended the config parameter auth to support `managed_identity` and `default_credentials`. Also fix the issue where the errors were being ignored.

Link to tracking Issue: 
1) Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31268
2) No tracing issues for extenting auth methods

Testing: Tested on my Azure VMs with managed identity

Documentation: Added to README.md

### CODE REVIEW

**Comment by nslaughter** [comment]
The main intention of the PR looks great and the implementation is consistent with the existing code. CI should be triggered now.

**Comment by nslaughter** [comment]
@codeboten are you able to approve the pending workflows?

File: .chloggen/azuremonitor-additional-authentication.yaml
Comment (nslaughter) [inline .chloggen/azuremonitor-additional-authentication.yaml:12]:
```suggestion
note: Add support for Managed Identity and Default Credential auth
```

Comment (sriniketh923) [reply .chloggen/azuremonitor-additional-authentication.yaml]: Yes updated it

**Comment by sriniketh923** [comment]
@codeboten @nslaughter can you please review again?

**Review by nslaughter — APPROVED** [review]
Thank you for the contribution and responses. LGTM

File: receiver/azuremonitorreceiver/scraper.go
Comment (sriniketh923) [reply receiver/azuremonitorreceiver/scraper.go:143]: Yes of course!

**Comment by sriniketh923** [comment]
@codeboten anything else you would like me to do before you can merge this?

**Comment by brendtg** [comment]
@codeboten , any update on this? Or a timeline we can expect for a merge?

**Review by crobert-1 — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**
